app/assets/javascripts/discourse/app/static/wizard/components/fields/radio.gjs

```diff
@@ -0,0 +1,75 @@
+import Component from "@glimmer/component";
+import { Input } from "@ember/component";
+import { on } from "@ember/modifier";
+import { action, set } from "@ember/object";
+import { hash } from "@ember/helper";
+import PluginOutlet from "discourse/components/plugin-outlet";
+import icon from "discourse-common/helpers/d-icon";
+
+export default class Radio extends Component {
+  constructor() {
+    super(...arguments);
+
+    this._setSelected();
+  }
+
+  get field() {
+    return this.args.field;
+  }
+
+  @action
+  selectionChanged(input) {
+    this.field.value = input.target.value;
+    this._resetSelected();
+    this._setSelected();
+  }
+
+  _resetSelected() {
+    for (let choice of this.field.choices) {
+      set(choice, "selected", false);
+    }
+  }
+
+  _setSelected() {
+    for (let choice of this.field.choices) {
+      if (this.field.value === choice.id) {
+        set(choice, "selected", true);
+      }
+    }
+  }
+
+  <template>
+    <div class="wizard-container__radio-choices">
+      {{#each @field.choices as |c|}}
+        <div class="wizard-container__radio-choice {{if c.selected 'selected'}}">
+          <label class="wizard-container__label">
+            <PluginOutlet
+              @name="wizard-radio"
+              @outletArgs={{hash disabled=c.disabled}}
+            >
+              <Input
```

Comment:
Should use native `input` here rather than ember's `Input`. You need to use `on "change"` and `withEventValue` instead of the click event. Here is a good example of radio from FK:

https://github.com/discourse/discourse/blob/d11d7158538978f7b9f809c6a5f04fdda48af7a9/app/assets/javascripts/discourse/app/form-kit/components/fk/control/radio-group/radio.gjs#L11-L21

Reply:
`withEventValue` is such a neat helper. 🥹